File: questions/1/python/1.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prev_nums = {}

        for idx, num in enumerate(nums):
            new_target = target - num

            if new_target in prev_nums:
                return prev_nums[new_target], idx

            prev_nums[num] = idx

        return [-1, -1]

File: questions/1/java/1.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> prevNums = new HashMap<>();

        for (int idx = 0; idx < nums.length; ++idx) {
            int newTarget = target - nums[idx];

            if (prevNums.containsKey(newTarget)) {
                return new int[] {prevNums.get(newTarget), idx};
            }

            prevNums.put(nums[idx], idx);
        }

        return new int[] {-1, -1};
    }
}